### PR TITLE
[DX-2357] Send engine and platform info on Passport init

### PR DESCRIPTION
**Pull request by CodeSchwert**
# Summary

[DX-2357](https://immutable.atlassian.net/browse/DX-2357)

Send engine version and platform info from Passport init to game bridge.

[DX-2357]: https://immutable.atlassian.net/browse/DX-2357?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by nattb8 — APPROVED** [review]

**Review by nattb8 — COMMENTED** [review]
No need to update the index file in the repo?

**Review by dom-murray — DISMISSED** [review]

**Review by carleeto — DISMISSED** [review]

File: src/Packages/Passport/Runtime/Scripts/Private/PassportImpl.cs [hunk 1]
```diff
@@ -36,7 +40,17 @@ public PassportImpl(IBrowserCommunicationsManager communicationsManager)
         public async UniTask Init(string clientId, string environment, string? redirectUri = null, string? deeplink = null)
         {
             this.redirectUri = redirectUri;
-            InitRequest request = new() { ClientId = clientId, Environment = environment, RedirectUri = redirectUri };
+
+            var versionInfo = new Dictionary<string, string>
```
Comment (nattb8) [inline src/Packages/Passport/Runtime/Scripts/Private/PassportImpl.cs:44]: Should we just create a class for this version info? Rather than a dictionary?

Comment (CodeSchwert) [reply src/Packages/Passport/Runtime/Scripts/Private/PassportImpl.cs:44]: Would that be preferable over a Dictionary?

Comment (nattb8) [reply src/Packages/Passport/Runtime/Scripts/Private/PassportImpl.cs:44]: Personally, I prefer it more than magic strings. Or at least define the keys as a constant at the top of the class.

Comment (CodeSchwert) [reply src/Packages/Passport/Runtime/Scripts/Private/PassportImpl.cs]: Moved the version info stuff to a separate class

**Review by carleeto — DISMISSED** [review]

File: src/Packages/Passport/Runtime/Scripts/Private/VersionInfo.cs [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        static VersionInfo()
+        {
+            UnityVersion = Application.unityVersion;
+            Platform = Application.platform;
+            OsVersion = SystemInfo.operatingSystem.Replace(" ", "_");
+        }
+
+        public Dictionary<string, string> GetVersionInfo()
+        {
+            Dictionary<string, string> versionInfo = new Dictionary<string, string>
```
Comment (nattb8) [inline src/Packages/Passport/Runtime/Scripts/Private/VersionInfo.cs:21]: To clarify my previous comment, I suggested creating a model class to hold the version information. Not moving the code into a different class.

```
[JsonObject(NamingStrategyType = typeof(CamelCaseNamingStrategy))]
public class VersionInfo
  {
      public string Engine;
      public string UnityVersion;
      public string Platform;
     public string PlatformVersion;
  }
```

Or keep it as is in the original file but store the magic strings (e.g. `"engine"`, `"engineVersion"`) as constants at the top.

```
private const string KEY_ENGINE = "engine";
private const string KEY_ENGINE_VERSION = "engineVersion";
...
```

**Review by nattb8 — APPROVED** [review]

**Review by carleeto — APPROVED** [review]

**Pull request merged**